src/hw/raspi4/libs/string.c

```diff
@@ -0,0 +1,90 @@
+/**
+ * We made our own implementation here because
+ * toolchain-provided implementation contains
+ * some instructions that has a strict rule to use.
+ *
```

Comment:
FYI, I've been looking at building without the clib in general. (we build "freestanding" but add `-lc` to the link at the moment)

So don't change anything now but I might move this later into common/string.c and use it for all the ports. (compiled with `-mstrict-align` so that it still works for the rpi4)